MITLibraries/solenoid
# Data import ui

#### What does this PR do?
Makes all of the UI changes (and corresponding tests) to allow for complete data import from Elements via API. Closes #215, #216.

#### How can a reviewer see these changes?
Due to IP-address-based access restrictions on the Elements API, you can’t test this in the PR build on Heroku. But you _can_ test it locally, you just have to be on campus or on the VPN.

To test locally:
- Checkout the branch, make sure the following env variables are set in your .env:
    - ALLOWED_HOSTS=localhost or DJANGO_DEBUG=True (or both)
    - DJANGO_SECRET_KEY=[whatever]
    - DJANGO_EMAIL_TESTING_MODE=True
    - DJANGO_ELEMENTS_USER=[in Heroku staging env, or ask Helen]
    - DJANGO_ELEMENTS_PASSWORD=[in Heroku staging env, or ask Helen]
- `pipenv shell` then `python manage.py runserver` to run the app. You will also need to run a separate Redis instance and Celery worker to test the Elements update process (not part of this PR, so not necessary unless you want to).
- Navigate to localhost:8000 and go to the Data import page. You should see an author ID box. Copy/paste into this an author ID from the Elements dev instance (note, these are not MIT IDs, but internal Elements IDs and are thus specific to the dev instance for testing). You can either go to the dev instance and find a faculty author to use (if you want to see the articles there for yourself and confirm they are indeed being imported), or pick from one of these: 12605, 12375, 12444, 10978.
- Import and wait for the import process to finish. You should see some messages about articles imported. Feel free to try re-importing an author to see those messages too. This is the end of the work covered by this PR, but…
- You can also complete the process of generating and sending emails if you want to see the whole process happen (note, emails will not actually send as long as DJANGO_EMAIL_TESTING_MODE is True). After “sending” an email you’ll see a message that Elements will be updated within the hour, which also will not actually happen unless you have a Celery worker running.

#### Includes new or updated dependencies?
YES (just updating Django bc the PEP safety check was failing, not related to the new functionality)

#### Reviewer checklist
- [ ] There are tests covering any new functionality
- [ ] The changes, if applicable, have been verified
